src/platform/packages/shared/kbn-apm-synthtrace/src/cli/utils/stream_manager.ts

```diff
@@ -98,21 +99,34 @@ export class StreamManager {
     generator: SynthGenerator<Fields>
   ): Promise<void> {
     const clientStream = this.createOrReuseClientStream(client);
+    const generatorStream = new PassThrough({ objectMode: true });
+    const generatorArray = castArray(generator);
+
+    generatorArray.reverse().forEach((current) => {
+      const currentStream = isGeneratorObject(current) ? Readable.from(current) : current;
+      currentStream.pipe(generatorStream, { end: false });
+      this.trackedStreams.push(currentStream);
```

Comment:
should this be local state?

Reply:
This is a valid concern with multiple clients since `generatorStream` is a local variable. I'll look into this.

Reply:
[7859020](https://github.com/elastic/kibana/pull/215291/commits/78590207b25d4edb0fbd42cf912f2a4f642c76be)